python/mxnet/gluon/data/text/base.py

```diff
@@ -0,0 +1,89 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+# pylint: disable=
+
+"""Base classes for text datasets and readers."""
+
+__all__ = ['TextTokenReader']
+
+import io
+import os
+
+from ..dataset import Dataset
+from ..datareader import DataReader
+
+class TextTokenReader(DataReader):
+    """Text reader that produces lists of tokens.
```

Comment:
why combine tokenizer with reader?

Reply:
For this dataset, word tokenization happens in the data reader because the result of tokenization could decide the sample boundary. It's also possible to defer the chunking-by-length to the dataset class. I will change it to the later and see how it is.